src/omnisharp/launcher.ts

```diff
@@ -189,34 +183,31 @@ function launch(details: LaunchDetails): Promise<LaunchResult> {
 }
 
 function launchWindows(details: LaunchDetails): Promise<LaunchResult> {
-    return new Promise<LaunchResult>(resolve => {
-
-        function escapeIfNeeded(arg: string) {
-            const hasSpaceWithoutQuotes = /^[^"].* .*[^"]/;
-            return hasSpaceWithoutQuotes.test(arg)
-                ? `"${arg}"`
-                : arg;
-        }
+    function escapeIfNeeded(arg: string) {
+        const hasSpaceWithoutQuotes = /^[^"].* .*[^"]/;
+        return hasSpaceWithoutQuotes.test(arg)
+            ? `"${arg}"`
+            : arg;
+    }
 
-        let args = details.args.slice(0); // create copy of details.args
-        args.unshift(details.serverPath);
-        args = [[
-            '/s',
-            '/c',
-            '"' + args.map(escapeIfNeeded).join(' ') + '"'
-        ].join(' ')];
-
-        let process = spawn('cmd', args, <any>{
-            windowsVerbatimArguments: true,
-            detached: false,
-            cwd: details.cwd
-        });
+    let args = details.args.slice(0); // create copy of details.args
+    args.unshift(details.serverPath);
+    args = [[
+        '/s',
+        '/c',
+        '"' + args.map(escapeIfNeeded).join(' ') + '"'
+    ].join(' ')];
+
+    let process = spawn('cmd', args, <any>{
+        windowsVerbatimArguments: true,
+        detached: false,
+        cwd: details.cwd
+    });
 
-        return resolve({
-            process,
-            command: details.serverPath,
-            usingMono: false
-        });
+    return Promise.resolve({
+        process,
+        command: details.serverPath,
+        usingMono: false
     });
```

Comment:
Why not putting all the code in a Promise like previously?
Any error there will end up in a throw which is not really what a function returning a promise should do.